improver/threshold.py

```diff
@@ -342,6 +357,7 @@ def process(self, input_cube: Cube) -> Cube:
 
             self._add_threshold_coord(cube, threshold)
             cube.coord(var_name="threshold").convert_units(input_cube.units)
+            self._update_metadata(cube, update_cell_methods=False)
```

Comment:
Do you need to update the cube name twice, or could you split this function up to make it clearer what's being updated each time you call it?

Reply:
I need to check I understand what you are after.

Are you suggesting I split out the setting of the name, cell method, and the rest (grouping cube units, coordinate name and attributes together) into three separate functions and call each one only where required?

I would then need to call all three in sequence in `LatitudeDependentThreshold` which also uses this function, or I could write a wrapper for cases where we want to call them all (but that's not so different from where it is right now).

Reply:
This is no longer relevant